woodwork/accessor_utils.py

```diff
@@ -82,10 +85,11 @@ def _update_column_dtype(series, logical_type):
                     series = pd.to_datetime(series, format=logical_type.datetime_format)
             else:
                 series = series.astype(new_dtype)
+                if series.dtype != new_dtype:
```

Comment:
Probably worth doing `if str(series.dtype) != new_dtype` to avoid potential weirdness incomparing a dtype object to a string

Reply:
Updated!